### ISSUE TITLE
Did you succeed？

### ISSUE THREAD

**Issue by lishengyuan** [redcweed/z490#1]
I also use the MSI GAMING EDGE motherboard, looking forward to sharing efi files.

**Comment by yukunyi** [comment]
I also use this motherboard, the CPU is i7 10700k, currently using 
`https://www.tonymacx86.com/threads/msi-z490-tomahawk-i7-10700k-catalina-15-5-5-opencore.299354/` 
this The EFI configuration in the post is successfully installed, but the network card cannot be driven, and the sound card cannot be driven. After entering the system, as long as the window is moved, this window will be white, and it will not operate like a crash.

**Comment by planetwally** [comment]
I am still awaiting the CPU (10700 arriving tomorrow hopefully) -- Once I get a working system I'll post my EFI and any relevant files. On my 9600k/z390 setup I found clover gave me slightly higher R20 results so I will be trying both Clover and OC and sticking with whichever works better. More coming very soon hopefully... Motherboard is here, but CPU was supposed to arrive Friday, but did not... @yukunyi have you tried the Lucy driver for the RT 2.5G ethernet?

**Comment by yukunyi** [comment]
[EFI.zip](https://github.com/redcweed/z490/files/4777527/EFI.zip)
As of now, the graphics card is working properly, the sound card has sound with VoodooHDA.kext, the network card has been driven with LucyRTL8125Ethernet.kext, and it can recognize the network card interface, but it has no response when plugged in, DHCP does not work, the network card light does not turn on, IntelBluetoothFirmware.kext Bluetooth after use The device does not work, ax201 wireless network card is not driven

**Comment by yukunyi** [comment]
R20 4601 points, lower than windows, windows ran 4778 points

**Comment by planetwally** [comment]
Interesting - thanks for sharing your EFI folder - as soon as I get the cpu tomorrow I will give things a try and I'll update with my EFI here. The wifi card I did not expect to work, but the ethernet should work with the Lucy kext... hmm it seems this board will be more challenging than I thought.

**Comment by yukunyi** [comment]
AppleALC ALCS1220A , I tested all layout id not work.

**Comment by planetwally** [comment]
hey @yukunyi  - what bios settings did you use? I just got the cpu and cannot get any EFI folder to work, tried yours also - i'm getting stuck at [EB|#LOG:EXITBS:START] ..

**Comment by yukunyi** [comment]
What version of MACOS installation image do you use?

**Comment by planetwally** [comment]
im trying with Catalina 15.4

**Comment by yukunyi** [comment]
I am using 10.15.5

**Comment by planetwally** [comment]
i'm going to try downloading 15.5 and see if that helps -- so far I have tried everything else :/

**Comment by yukunyi** [comment]
Do you have a discrete graphics card?

**Comment by planetwally** [comment]
yes i have a rx590 -- i tried both with the card installed and with the card uninstalled and same result... do you have any special BIOS settings ?

**Comment by yukunyi** [comment]
![20200616_094025](https://user-images.githubusercontent.com/1285369/84723795-125b3800-afb9-11ea-8e28-af263588b8bf.jpg)
![20200616_094045](https://user-images.githubusercontent.com/1285369/84723801-14bd9200-afb9-11ea-8d3c-cc86fe306119.jpg)
![20200616_094052](https://user-images.githubusercontent.com/1285369/84723804-15eebf00-afb9-11ea-828f-8aa7d30557dc.jpg)
![20200616_094059](https://user-images.githubusercontent.com/1285369/84723809-17b88280-afb9-11ea-90d2-e349bd66e0a2.jpg)
![20200616_094124](https://user-images.githubusercontent.com/1285369/84723811-18e9af80-afb9-11ea-810c-7b1da71731ba.jpg)

**Comment by yukunyi** [comment]
[EFI.zip](https://github.com/redcweed/z490/files/4783936/EFI.zip)
This is the configuration I currently use. I added RX580, your RX590 should also be drive-free, and the discrete graphics card will be automatically configured.

**Comment by planetwally** [comment]
thank you im going to try to match all my bios settings to yours and use your EFI also. I'm also downloading catalina 15.5 to see if that helps ... thank you -- also do you use nvme m.2 drive?

**Comment by yukunyi** [comment]
I am using Samsung 970 evo plus. MZ-V7S500B

**Comment by planetwally** [comment]
our setup is very similar - mine is a z490m + 10700 (non-k) + wd black 500 nvme m.2 + rx 590 -- I just checked and my bios is the same settings as yours. Im not sure why it refuses to boot install disk or my preinstalled ssd from my z390 machine. I'm still waiting on the download 15.5 -- hoping that will resolve the issue.

**Comment by yukunyi** [comment]
Where did you download 10.15.5?

**Comment by planetwally** [comment]
from the App Store on my macbook -- you can search for it in the app store if you have any real mac and it will allow you to download it.

**Comment by yukunyi** [comment]
I downloaded it through the softwareupdate command from my macbook pro.

**Comment by planetwally** [comment]
yup - that is basically same - mine has 10 more minutes to go then i'll get to try it with 15.5 ... if that doesn't work then I really don't know what else to try...

**Comment by planetwally** [comment]
well ... no luck :( i tried your EFI folder, Catalina 15.5, same BIOS settings as you, yet i am still getting same error :(

**Comment by planetwally** [comment]
i have tried everything now ... removed the gpu, tried from igpu, tried from rx590, removed m.2, tried different usb sticks, tried clearing cmos many times, tried all different bios settings ... same error ....

**Comment by yukunyi** [comment]
I can only say, come on.

**Comment by yukunyi** [comment]
Is there any progress?

**Comment by planetwally** [comment]
i managed to get it to boot with a patch from another EFI - same as you Im starting to troubleshoot the network adapter - it appears, but does not recognize a connection. going to be testing today and hopefully sort it out - will post as soon as I make any developments

**Comment by planetwally** [comment]
I'm making some progress on the ethernet - try forcing it into 1000BaseT mode and then use DHCP with manual IP -- I am able to get a connection to my router and have light on rear of moboard ethernet port.

**Comment by planetwally** [comment]
To force 1g - go to Ethernet -> Advanced -> Hardware -> Speed 1000baseT

**Comment by yukunyi** [comment]
At present, I am using a type-c converter with a gigabit network card free of drive, but occasionally there will be some abnormalities on the network. It is temporarily unclear whether it is related to the system, maybe it is just a bug in my program~ ~~

**Comment by yukunyi** [comment]
Are you using the LucyRTL8125Ethernet module, force 1G rate, and then manually configure the IP?

**Comment by yukunyi** [comment]
At present, I have been working with him for 2 days. I very much hope that AppleALC can work normally. At present, I can output sound normally through the audio port of AppleALC display.

**Comment by planetwally** [comment]
yes i am using LucyRTL8125Ethernet -- however even when i manually configure IP - while it says it is connected, lights are flashing, etc, i am still not getting network ... will keep trying until i get it working

**Comment by planetwally** [comment]
sound was not working for me with AppleALC so i replaced it with VoodooHDA which appears to be working perfectly for me - the only issue i have is changing volume is laggy and spikes the CPU for some reason

**Comment by planetwally** [comment]
I got it working - i'll upload the updated Lucy driver to my repo - please try that one along with forcing the 1000BaseT (or 100 if that is what your router supports) - it now works perfectly.

**Comment by planetwally** [comment]
@yukunyi did you try the Lucy kext i uploaded ? try it out - works perfectly now

**Comment by yukunyi** [comment]
@redcweed In your way, my onboard network card is working fine. I am currently experiencing another problem, that is, vmware fusion cannot mount USB devices to the virtual machine

**Comment by yukunyi** [comment]
Not all USB devices will work, some will. Not necessarily caused by black apples.

**Comment by planetwally** [comment]
Try the USBSerie400.kext instead of USBInjectAll.kext --
https://github.com/redcweed/z490/tree/master/EFI/OC/Kexts
